src/backend/dx12/src/command.rs

```diff
@@ -1231,17 +1293,231 @@ impl com::RawCommandBuffer<Backend> for CommandBuffer {
+                num_layers as _,
+            );
+            self.rtv_pools.push(rtv_pool.raw.clone());
+
+            let blit = match r.dst_subresource.aspects {
+                format::Aspects::COLOR => {
+                    // Create RTVs of the dst image for the miplevel of the current region
+                    for i in 0 .. num_layers {
+                        let mut desc = d3d12::D3D12_RENDER_TARGET_VIEW_DESC {
+                            Format: dst.dxgi_format,
+                            ViewDimension: d3d12::D3D12_RTV_DIMENSION_TEXTURE2DARRAY,
+                            u: unsafe { mem::zeroed() },
+                        };
+
+                        *unsafe { desc.u.Texture2DArray_mut() } = d3d12::D3D12_TEX2D_ARRAY_RTV {
+                            MipSlice: r.dst_subresource.level as _,
+                            FirstArraySlice: (i + first_layer) as u32,
+                            ArraySize: 1,
+                            PlaneSlice: 0, // TODO
+                        };
+
+                        let view = rtv_pool.at(i as _).cpu;
+                        unsafe {
+                            device.CreateRenderTargetView(dst.resource, &desc, view);
+                        }
+                    }
+
+                    self.shared.service_pipes.get_blit_2d_color(dst.dxgi_format, filter)
+                },
+                _ => unimplemented!(),
+            };
+
+            self.set_internal_graphics_pipeline();
+            // TODO: reset descriptor heaps
+
+            // Take flipping into account
+            let viewport = d3d12::D3D12_VIEWPORT {
+                TopLeftX: cmp::min(r.dst_bounds.start.x, r.dst_bounds.end.x) as _,
+                TopLeftY: cmp::min(r.dst_bounds.start.y, r.dst_bounds.end.y) as _,
+                Width: (r.dst_bounds.end.x - r.dst_bounds.start.x).abs() as _,
+                Height: (r.dst_bounds.end.y - r.dst_bounds.start.y).abs() as _,
+                MinDepth: 0.0,
+                MaxDepth: 1.0,
+            };
+            let scissor = d3d12::D3D12_RECT {
+                left: viewport.TopLeftX as _,
+                top: viewport.TopLeftY as _,
+                right: (viewport.TopLeftX + viewport.Width) as _,
+                bottom: (viewport.TopLeftY + viewport.Height) as _,
+            };
+
+            let flip_x = (r.dst_bounds.end.x - r.dst_bounds.start.x) < 0;
+            let flip_y = (r.dst_bounds.end.y - r.dst_bounds.start.y) < 0;
+
+            unsafe {
+                self.raw.RSSetViewports(1, &viewport);
+                self.raw.RSSetScissorRects(1, &scissor);
+                self.raw.IASetPrimitiveTopology(d3dcommon::D3D_PRIMITIVE_TOPOLOGY_TRIANGLELIST);
+                self.raw.SetPipelineState(blit.pipeline.as_raw());
+                self.raw.SetGraphicsRootSignature(blit.signature.as_raw());
+                self.raw.SetGraphicsRootDescriptorTable(0, srv_heap_gpu);
+            }
+
+            let mut barriers = (0 .. num_layers)
```

Comment:
The barriers should be moved out of the region loops.
It might be possible that two regions access the same subresource, messing with the resource states. Therefore barriers should be set once on begin of the blitcommand and once at the end.
(The worst case would be if src and dst image are the same but not sure if this is allowed..)

Reply:
That's ideal but more problematic to actually implement...